### DISCUSSION TITLE
How DB Scheduler works in multiple instance environment?

### DISCUSSION THREAD

**Discussion by ashwganes** [kagkarlsson/db-scheduler#440] [Q&A]
I want to understand how DB scheduler works in multi instance environment. If a brief explanation with an example where application is deployed in 3 instances, is provided, it would help a lot.
Above which, I will try to get a deeper understanding.

**Comment by ashwganes** [comment]
Can someone please throw some light on this?
It will be really helpful.
Thanks in advance.

**Comment by ashwganes** [comment]
Can someone please respond on this, as it will be helpful for me take the next steps

**Comment by kagkarlsson** [comment]
Sorry, limited time. Short answer is: it works. Two different polling strategies. One uses optimistic locking to ensure it gets an exclusive "lock" (record marked as "picked"), the other uses a select-for-update and marks the record as "picked". As long as it is picked, no other instance will try to run it. 
The only thing that can happen is that heartbeats fail for a long time, then it is considered "dead", and unlocked/unpicked (unless configured to do otherwise)
